### ISSUE TITLE
class-by-class IOU metric plot for multi-class

### ISSUE THREAD

**Issue by rak5216** [mit-quest/necstlab-damage-segmentation#41]
Separate curves on same plot would be ideal for IOU.

**Comment by rak5216** [comment]
branch `rak/resolve_issue41_class_metrics` addresses this

**Comment by rak5216** [comment]
Global metrics (incorrect if computed per batch): precision, recall, f score (& dice loss), IOU (& jaccard loss)
Batch-capable metrics (correct for global with running avg of batch): accuracy

https://datascience.stackexchange.com/questions/45165/how-to-get-accuracy-f1-precision-and-recall-for-a-keras-model

**Comment by rak5216** [comment]
Metrics/losses from Keras vs. segmentation_models vs. custom:

[Keras metrics](https://github.com/keras-team/keras/blob/7a39b6c62d43c25472b2c2476bd2a8983ae4f682/keras/metrics.py#L1891): 
- accuracy:
- binary_crossentropy:
- categorical_crossentropy:

segmentation_models ([metrics ](https://github.com/qubvel/segmentation_models/tree/9e8dfec9ba3a67a8884eb128e05f3e126407f6b5/segmentation_models/base)and [losses](https://github.com/qubvel/segmentation_models/blob/master/segmentation_models/losses.py)):
- IOUScore
- FScore
- Precision
- Recall
- JaccardLoss
- DiceLoss
- CategoricalCELoss

Custom metrics:
- ClassBinaryAccuracy

**Comment by rak5216** [comment]
For all metrics (and their dependent loss fcns, if they exist) except for binary_accuracy and categorical accuracy from keras, the current issue is that 1-hot ground truth labels are compared to softmax predictions (0<= pred <=1). Either need to threshold all predictions or apply hardmax as in infer, so that predictions become 1-hot also prior to metric calculation. Note, no effect on cross-entropy.

**Comment by rak5216** [comment]
for separate class curves on same plot, do post process with csv metric output
